catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/ConfigurationStore.java

```diff
@@ -78,9 +92,51 @@ public boolean isInMemory() {
 
   public void setInMemory(boolean isInMemory) {
     inMemory = isInMemory;
+
+    notifyListeners();
   }
 
   public void setForceAutoCommit(boolean forceAutoCommit) {
     this.forceAutoCommit = forceAutoCommit;
   }
+
+  public void setAnyTextFieldWhitelist(List<String> fieldWhitelist) {
+    this.anyTextFieldWhitelist.clear();
```

Comment:
:question: The blueprint provide list is thread safe.  Is there another reason you are clearing and adding the elements?

Reply:
This pattern makes the list completely owned and managed by the ConfigurationStore.  Even though the blueprint list is thread safe if I used the list provided as a parameter I couldn't guarantee that the ConfigurationStore would know about everything that was added to (or removed from) the list so that it could properly notify the listeners of the change.  In theory, if I didn't do this, someone could pass in a list and then modify the list (add or remove) without ever causing a notification to get to the DynamicSchemaResolver.  While this won't happen with the current setup, I felt it was a better design.